src/datePickerAndroid.re

```diff
@@ -0,0 +1,57 @@
+[@bs.scope "DatePickerAndroid"] [@bs.module "react-native"] external dateSet : string =
+  "dateSetAction";
+
+[@bs.scope "DatePickerAndroid"] [@bs.module "react-native"] external dismissed : string =
+  "dismissedAction";
+
+type mode =
+  | Calendar
+  | Spinner
+  | Default;
+
+type response = {
+  year: int,
+  month: int,
+  day: int
+};
+
+type action =
+  | Dismissed
+  | Set(response);
+
+let action = (resp) =>
+  resp##action == dateSet ?
+    Set({year: resp##year, month: resp##month, day: resp##day}) : Dismissed;
+
+type responseJs = {. "action": string, "year": int, "month": int, "day": int};
+
+type optsJs = {
+  .
+  "date": Js.Date.t,
+  "minDate": Js.Undefined.t(Js.Date.t),
+  "maxDate": Js.Undefined.t(Js.Date.t),
+  "mode": string
+};
+
+[@bs.scope "DatePickerAndroid"] [@bs.module "react-native"]
+external _open : optsJs => Js.Promise.t(responseJs) =
+  "open";
+
+let open_ = (~date: Js.Date.t, ~minDate=?, ~maxDate=?, ~mode=Default, ()) =>
+  _open({
+    "date": date,
+    "minDate": Js.Undefined.from_opt(minDate),
```

Comment:
I'd go with `Js.Nullable` here it covers both null and undefined but the from_opt value get's compiled down to `undefined`

Reply:
I'm also pretty sure `Js.Undefined` and `Js.Null` are getting deprecated in favour of `Js.Nullable` soon